core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -736,6 +748,12 @@ public void updateVersion(long version) {
             this.doc.version().setLongValue(version);
         }
 
+        @Override
+        public void updateSeqNo(long seqNo) {
+            super.updateSeqNo(seqNo);
+            this.doc.seqNo().setLongValue(seqNo);
```

Comment:
I followed the same pattern the version field takes. We should re-evaluate both if we're not happy with it. I'll have a second look at the implications. I presume you mean the lucene field should be immutable?